src/GafferScene/ImageScatter.cpp

```diff
@@ -0,0 +1,371 @@
+		V2f( displayWindow.min.x * pixelAspect, displayWindow.min.y ),
+		V2f( displayWindow.max.x * pixelAspect, displayWindow.max.y )
+	);
+
+	// Generate positions using a PointDistribution reading from a Sampler for
+	// the density channel.
+
+	Sampler densitySampler( imagePlug(), densityChannel, displayWindow, Sampler::Clamp );
+	densitySampler.populate(); // Multithread the population of image tiles
+
+	// Point distribution is designed for samping within a unit square, so we
+	// offset and scale to fit that to our input image.
+	const float scale = std::max( outputArea.size().x, outputArea.size().y );
+	const V2i offset = displayWindow.min;
+
+	auto densityFunction = [&] ( const V2f &p ) {
+		IECore::Canceller::check( context->canceller() );
+		return densitySampler.sample( offset.x + p.x * scale / pixelAspect, offset.y + p.y * scale );
+	};
+
+	V3fVectorDataPtr positionsData = new V3fVectorData;
+	positionsData->setInterpretation( IECore::GeometricData::Point );
+	vector<V3f> &positions = positionsData->writable();
+	auto emitter = [&] ( const V2f &p ) {
+		positions.push_back( V3f( offset.x + p.x * scale, offset.y + p.y * scale, 0.0f ) );
+	};
+
+	/// \todo It would be nice to multithread this, but it could also be pretty
+	/// handy that the order of the points we're outputting matches the
+	/// progressive order in which they are generated.
+	PointDistribution::defaultInstance()(
+		Box2f( V2f( 0 ), V2f( outputArea.size() ) / scale ),
+		// Scale density to be in points per pixel
+		densityPlug()->getValue() * scale * scale,
+		densityFunction,
+		emitter
+	);
+
+	// Make a PointsPrimitive from the positions
+
+	PointsPrimitivePtr result = new PointsPrimitive( positionsData );
+
+	// Add on primitive variables.
+
+	const float width = widthPlug()->getValue();
+	if( widthChannel.empty() )
+	{
+		result->variables["width"] = PrimitiveVariable( PrimitiveVariable::Interpolation::Constant, new FloatData( width ) );
+	}
+
+	const std::string primitiveVariablesMatchPattern = primitiveVariablesPlug()->getValue();
+	for( const auto &channelName : channelNamesData->readable() )
+	{
+		if( StringAlgo::matchMultiple( channelName, primitiveVariablesMatchPattern ) )
+		{
+			const int colorIndex = ImageAlgo::colorIndex( channelName );
+			if( colorIndex >= 0 && colorIndex <= 2 )
+			{
+				// Map R, G and B to the components of colour primitive variables.
+				// This is the same behaviour as ImageToPoints.
+				string name = ImageAlgo::layerName( channelName );
+				name = name == "" ? "Cs" : name;
+				Color3fVectorDataPtr colorData = result->variableData<Color3fVectorData>( name );
+				if( !colorData )
+				{
+					colorData = new Color3fVectorData;
+					colorData->writable().resize( positions.size() );
+					result->variables[name] = PrimitiveVariable( PrimitiveVariable::Vertex, colorData );
+				}
+				sampleChannel( imagePlug(), displayWindow, channelName, positions, context->canceller(), colorData->baseWritable() + colorIndex, 3 );
```

Comment:
Do we need to account for the pixel aspect here and in other use of `sampleChannel`? I'm seeing some odd behaviour sampling colours, widths, etc. from non-square pixel images.

![image](https://github.com/GafferHQ/gaffer/assets/50844517/3d0e5ed4-2e7b-4768-8831-877aa615de30)

```
import Gaffer
import GafferImage
import GafferScene
import imath

Gaffer.Metadata.registerValue( parent, "serialiser:milestoneVersion", 1, persistent=False )
Gaffer.Metadata.registerValue( parent, "serialiser:majorVersion", 3, persistent=False )
Gaffer.Metadata.registerValue( parent, "serialiser:minorVersion", 4, persistent=False )
Gaffer.Metadata.registerValue( parent, "serialiser:patchVersion", 0, persistent=False )

__children = {}

__children["ImageScatter"] = GafferScene.ImageScatter( "ImageScatter" )
parent.addChild( __children["ImageScatter"] )
__children["ImageScatter"].addChild( Gaffer.V2fPlug( "__uiPosition", defaultValue = imath.V2f( 0, 0 ), flags = Gaffer.Plug.Flags.Default | Gaffer.Plug.Flags.Dynamic, ) )
__children["Checkerboard"] = GafferImage.Checkerboard( "Checkerboard" )
parent.addChild( __children["Checkerboard"] )
__children["Checkerboard"].addChild( Gaffer.V2fPlug( "__uiPosition", defaultValue = imath.V2f( 0, 0 ), flags = Gaffer.Plug.Flags.Default | Gaffer.Plug.Flags.Dynamic, ) )
__children["ImageScatter"]["image"].setInput( __children["Checkerboard"]["out"] )
__children["ImageScatter"]["density"].setValue( 0.10000000149011612 )
__children["ImageScatter"]["primitiveVariables"].setValue( '*' )
Gaffer.Metadata.registerValue( __children["ImageScatter"]["primitiveVariables"], 'channelMaskPlugValueWidget:custom', True )
__children["ImageScatter"]["__uiPosition"].setValue( imath.V2f( 9.73065758, -17.4159966 ) )
__children["Checkerboard"]["format"].setValue( GafferImage.Format( 1080, 1080, 2.000 ) )
Gaffer.Metadata.registerValue( __children["Checkerboard"]["format"], 'formatPlugValueWidget:mode', 'custom' )
__children["Checkerboard"]["size"]["x"].setValue( 440.0 )
__children["Checkerboard"]["size"]["y"].setInput( __children["Checkerboard"]["size"]["x"] )
__children["Checkerboard"]["colorA"].setValue( imath.Color4f( 1, 0.25, 0, 1 ) )
__children["Checkerboard"]["colorB"].setValue( imath.Color4f( 0, 0.5, 0.5, 1 ) )
__children["Checkerboard"]["__uiPosition"].setValue( imath.V2f( 9.73065758, -9.25193405 ) )


del __children
```